packages/yourloops/test/unit/components/dialogs/medical-report-edit-dialog.hook.test.tsx

```diff
@@ -186,4 +208,148 @@ describe('MedicalReportEditDialog hook', () => {
+      const { result } = renderHook(() => useMedicalReportEditDialog({
+        medicalReport,
+        onSaved,
+        teamId,
+        patientId
+      }))
+      await act(async () => {
+        await result.current.saveMedicalReport()
+      })
+
+      expect(updateMedicalReportSpy).toHaveBeenCalledWith(medicalReport)
+      expect(successMock).toHaveBeenCalled()
+      expect(onSaved).toHaveBeenCalled()
+    })
+
+    it('should show error when update failed', async () => {
+      (authHookMock.useAuth as jest.Mock).mockImplementation(() => ({
+        user: {
+          id: 'fakeUserId',
+          isUserPatient: () => false,
+          isUserHcp: () => true
+        }
+      }))
+      const updateMedicalReportSpy = jest.spyOn(MedicalFilesApi, 'updateMedicalReport').mockRejectedValue(Error('This error was thrown by a mock on purpose'))
+      const { result } = renderHook(() => useMedicalReportEditDialog({
+        medicalReport,
+        onSaved,
+        teamId,
+        patientId
+      }))
+      await act(async () => {
+        await result.current.saveMedicalReport()
+      })
+
+      expect(updateMedicalReportSpy).toHaveBeenCalledWith(medicalReport)
+      expect(errorMock).toHaveBeenCalled()
+      expect(onSaved).not.toHaveBeenCalled()
+    })
+
+    it('should create successfully medical record', async () => {
+      const firstName = 'Sandy'
+      const lastName = 'Kilos';
+      (authHookMock.useAuth as jest.Mock).mockImplementation(() => ({
+        user: {
+          id: 'fakeUserId',
+          profile: { firstName, lastName },
+          isUserPatient: () => false,
+          isUserHcp: () => true
+        }
+      }))
+      const createMedicalReportSpy = jest.spyOn(MedicalFilesApi, 'createMedicalReport').mockResolvedValue(medicalReport)
+      const { result } = renderHook(() => useMedicalReportEditDialog({
+        onSaved,
+        teamId,
+        patientId
+      }))
+
+      await act(async () => {
+        result.current.setProgressionProposal(medicalReport.progressionProposal)
+        result.current.setDiagnosis(medicalReport.diagnosis)
+        result.current.setTrainingSubject(medicalReport.trainingSubject)
+      })
+
+      await act(async () => {
+        await result.current.saveMedicalReport()
+      })
+
+      const medicalReportExpected: NewMedicalReport = {
+        patientId,
+        teamId,
+        diagnosis: medicalReport.diagnosis,
+        progressionProposal: medicalReport.progressionProposal,
+        trainingSubject: medicalReport.trainingSubject,
+        authorFirstName: firstName,
+        authorLastName: lastName,
+        teamName: medicalReport.teamName
+      }
+
+      expect(createMedicalReportSpy).toHaveBeenCalledWith(medicalReportExpected)
+      expect(successMock).toHaveBeenCalled()
+      expect(onSaved).toHaveBeenCalled()
+    })
+
+    it('should show error when update failed', async () => {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (blocking):** ‏I think you meant "create" and not "update"? Now we have twice "should show error when update failed" which is not good... 😬